test/MUXControlsTestApp/VisualTreeTestBase.cs

```diff
@@ -0,0 +1,360 @@
+            {
+                Log.Comment(LogMasterFileRuntimeParameterName + ": True");
+            }
+            else
+            {
+                Log.Comment(LogMasterFileRuntimeParameterName + ": False");
+            }
+        }
+
+
+        private void VisualTreeCompare(UIElement root, string masterFilePrefix, string theme)
+        {
+            string content = "";
+            string expectedContent = "";
+
+            RunOnUIThread.Execute(() =>
+            {
+                content = VisualTreeDumper.DumpToXML(root, PropertyValueTranslator, VisualTreeDumpFilter);
+            });
+
+            MasterFileStorage storage = new MasterFileStorage(!ShouldLogMasterFile, masterFilePrefix, theme);
+            string bestMatchedMasterFileName = storage.BestMatchedMasterFileName;
+            string expectedMasterFileName = storage.ExpectedMasterFileName;
+
+            Log.Comment("Target master file: " + expectedMasterFileName);
+            Log.Comment("Best matched master file: " + bestMatchedMasterFileName);
+
+            VisualTreeOutputCompare result = new VisualTreeOutputCompare("", "");
+            if (String.IsNullOrEmpty(bestMatchedMasterFileName))
+            {
+                result.AddError("Can't find master file for " + TestCaseName);
+            }
+            else
+            {
+                expectedContent = MasterFileStorage.GetMasterFileContent(bestMatchedMasterFileName);
+                result = new VisualTreeOutputCompare(content, expectedContent);
+            }
+
+            if (result.HasError())
+            {
+                storage.LogMasterFile(expectedMasterFileName, content);
+                storage.LogMasterFile(expectedMasterFileName + ".orig", expectedContent);
+
+                Log.Comment(result.ToString());
+                string error = String.Format("Compare failed, but {0} is put into {1}", expectedMasterFileName, storage.StorageLocation);
+                Verify.Fail(error);
+            }
+        }
+
+        // Avoid too much logs, and only log message when !flag.
+        private void CheckTrue(bool flag, string message)
+        {           
+            if (!flag)
+            {
+                Verify.Fail(message);
+            }
+        }
+
+        private bool TestContextContainsKey(string key)
+        {
+#if USING_TAEF
+           return TestContext.Properties.Contains(key);
+#else
+            return TestContext.Properties.ContainsKey(key);
+#endif
+        }
+    }
+
+    class MasterFileStorage
+    {
+        private StorageFolder _storage;
+        public string StorageLocation { get; private set; }
+        public string ExpectedMasterFileName { get; private set; }
+        public string BestMatchedMasterFileName { get; private set; }
+        public MasterFileStorage(bool useLocalStorage, string masterFileNamePrefix, string theme)
+        {
+            _storage = useLocalStorage ? ApplicationData.Current.LocalFolder : KnownFolders.MusicLibrary;
```

Comment:
Why would this ever use the music library?

Reply:
After we run the test case, app is deleted. so we need to put master file into to a share folder. here I choose music folder.